community/dbms/src/main/java/org/neo4j/commandline/dbms/CsvImporter.java

```diff
@@ -54,29 +85,52 @@ public static String description()
                 "        http://neo4j.com/docs/operations-manual/current/deployment/#import-tool\n" +
                 "--input-encoding <character-set>\n" +
                 "        Character set that input data is encoded in. Defaults to UTF-8.\n" +
-                "--page-size <page-size>\n" +
-                "        Page size to use for import in bytes. (e. g. 4M or 8k)\n";
+                "--page-size <page-size>\n" + "        Page size to use for import in bytes. (e. g. 4M or 8k)\n";
     }
 
     public static String arguments()
     {
         return "[--nodes[:Label1:Label2]=\"<file1>,<file2>,...\"] " +
                 "[--relationships[:RELATIONSHIP_TYPE]=\"<file1>,<file2>,...\"] " +
-                "[--input-encoding=<character-set>] " +
-                "[--id-type=<id-type>] ";
+                "[--input-encoding=<character-set>] " + "[--id-type=<id-type>] ";
     }
 
-    private String[] args;
-
-    CsvImporter( String[] args, Config config )
+    CsvImporter( Args args, Config config, OutsideWorld outsideWorld ) throws IncorrectUsage
     {
-        this.args = Arrays.copyOf( args, args.length + 1 );
-        this.args[args.length] = String.format( "--into=%s", config.get( database_path ) );
+        this.args = args;
+        this.outsideWorld = outsideWorld;
+        nodesFiles = extractInputFiles( args, "nodes", outsideWorld.errorStream() );
+        relationshipsFiles = extractInputFiles( args, "relationships", outsideWorld.errorStream() );
+        try
+        {
+            validateInputFiles( nodesFiles, relationshipsFiles );
+        }
+        catch ( IllegalArgumentException e )
+        {
+            throw new IncorrectUsage( e.getMessage() );
+        }
+
+        idType = args.interpretOption( "id-type", withDefault( IdType.STRING ),
+                from -> IdType.valueOf( from.toUpperCase() ) );
+        inputEncoding = Charset.forName( args.get( "input-encoding", defaultCharset().name() ) );
+        pageSize = toIntExact( parseLongWithUnit( args.get( "page-size", String.valueOf( -1 ) ) ) );
+        this.config = config;
     }
 
+
     @Override
     public void doImport() throws IOException
     {
-        ImportTool.main( args );
+        BufferedOutputStream badOutput = new BufferedOutputStream( outsideWorld.errorStream() );
```

Comment:
Should flush the bufferedstream you construct here, since it can't be closed (in case errorStream is stdErr)

```
 BufferedOutputStream badOutput = new BufferedOutputStream( outsideWorld.errorStream() );
try {
    // ... all other code
} finally {
    badOutput.flush();
}